extensions/arc/src/ui/dashboards/miaa/miaaComputeAndStoragePage.ts

```diff
@@ -0,0 +1,371 @@
+		}).component();
+
+		const memoryVCoreslink = this.modelView.modelBuilder.hyperlink().withProperties<azdata.HyperlinkComponentProperties>({
+			label: loc.scalingCompute,
+			url: 'https://docs.microsoft.com/azure/azure-arc/data/configure-managed-instance',
+			CSSStyles: { 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p4 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartFour,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p5 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartFive,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p6 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartSix,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const computeInfoAndLinks = this.modelView.modelBuilder.flexContainer().withLayout({ flexWrap: 'wrap' }).component();
+		computeInfoAndLinks.addItem(infoComputeStorage_p1, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p2, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(memoryVCoreslink, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p4, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p5, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p6, { CSSStyles: { 'margin-right': '5px' } });
+		content.addItem(computeInfoAndLinks, { CSSStyles: { 'min-height': '30px' } });
+
+		this.configurationContainer = this.modelView.modelBuilder.divContainer().component();
+		this.configurationContainer.addItems(this.createUserInputSection(), { CSSStyles: { 'min-height': '30px' } });
+		content.addItem(this.configurationContainer, { CSSStyles: { 'margin-top': '30px' } });
+
+		this.initialized = true;
+
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// Save Edits
+		this.saveButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: loc.saveText,
+			iconPath: IconPathHelper.save,
+			enabled: false
+		}).component();
+
+		this.disposables.push(
+			this.saveButton.onDidClick(async () => {
+				this.saveButton!.enabled = false;
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: loc.updatingInstance(this._miaaModel.info.name),
+							cancellable: false
+						},
+						async (_progress, _token): Promise<void> => {
```

Comment:
what is this token parameter supposed to be for?

Reply:
CancellationToken

Reply:
do you think calling cancel on the token in the error path would be a useful thing to add?

Reply:
That's not what a cancellation token is for - it's for the task (callback) to be informed when the user requests cancellation. But since we specifically have cancellable set to false that case will never happen anyways.